checkpoint3/Program.cs

```diff
@@ -0,0 +1,257 @@
+                        choice = Console.ReadLine();
+                    }
+                    catch{
+                        Console.WriteLine();
+                        Console.WriteLine("try a different id");
+                        Console.WriteLine();
+                        Console.WriteLine("[press enter]");
+                        choice = Console.ReadLine();
+                    }
+                }
+                else if(choice == "list pending")
+                {
+                    cont.listPending();
+                    Console.WriteLine("[press enter]");
+                    choice = Console.ReadLine();
+                }
+                else if(choice == "list done")
+                {
+                    cont.listDone();
+                    Console.WriteLine("[press enter]");
+                    choice = Console.ReadLine();
+                }
+                else if(choice == "list all")
+                {
+                    cont.listAll();
+                    Console.WriteLine("[press enter]");
+                    choice = Console.ReadLine();
+                }
+                else
+                {
+                    if(choice != "quit")
+                    {
+                        Console.WriteLine("[ERROR] Invalid action");
+                        Console.WriteLine();
+                        Console.WriteLine("[press enter]");
+                        choice = Console.ReadLine();   
+                    }
+                }
+            }
+        }
+    }
+    class Controller
+    {
+        // public table todos_all {get; set;}
+        public Context context;
+        public Controller()     //DAO
+        {
+            // todos_all = new table();
+            context = new Context();
+            context.Database.EnsureCreated();
+
+        }
+        public List<table> list(){
+            List<table> theResult = new List<table>();
+            foreach(table aTable in context.todos){
+                theResult.Add(aTable);
+            }  
+            return theResult;
+        }
+        public void add(string item)    //(int id, string item)
+        {
+            // todos_all.id.Add(id);       //set int i in main that increases as one item is added
+            // todos_all.item.Add(item);
+            // todos_all.status.Add("pending");
+            context.todos.Add(new table(item));
+            context.SaveChanges();
+        }
+        public void delete(int itemId)
+        {
+            // todos_all.id.RemoveAt(itemId);
+            // todos_all.item.RemoveAt(itemId);
+            // todos_all.status.RemoveAt(itemId); 
+            context.todos.Remove(context.todos.Find(itemId));
+            context.SaveChanges();
+        }
+        public void markdone(int itemId)
+        {
+            // todos_all.status[itemId] = "done";
+            context.todos.Find(itemId).status = true;
+        }
+        public void listPending()
```

Comment:
your print methods have a lot of repeated code,  wha would be another way to solve it without having to repeat the code.